src/components/Map/MapIconButton.tsx

```diff
@@ -8,6 +9,40 @@ export interface IMapIconButtonProps {
   className?: string;
 }
 
+const MapIconUnimplemented = ({
+  children,
+  onClick,
+  type = "button",
+  className = "",
+}: IMapIconButtonProps) => {
+  return (
+    <Dialog.Root>
+      {/* Icon on map */}
+      <Dialog.Trigger asChild>
+        <button className={`${className} default-overlay`}>{children}</button>
+      </Dialog.Trigger>
+
+      {/* Appears when clicked */}
+      <Dialog.Portal>
+        <Dialog.Overlay />
+        <Dialog.Content className="fixed top-1/4 left-1/3 rounded-lg bg-white drop-shadow-xl p-8">
```

Comment:
Why isn't this centered? Can we use `top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2`

Reply:
I did play around with these for a little, and thought that this looked better; would love to hear other opinions though.
`top-1/4 left-1/3`:
![image](https://user-images.githubusercontent.com/57198797/219922471-fad3f0b3-fa0a-41d2-9f4e-a300f0a5a7b3.png)

`top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2`:
![image](https://user-images.githubusercontent.com/57198797/219922511-b4e96bfb-293f-4c9d-8323-aadd908a5698.png)

Reply:
The difference in positioning becomes more challenging as the screen size gets larger, I would pretty strongly push towards centering it. Also might be worth putting a global shadow when the modal is active.

Reply:
Centered. I don't think adding the global shadow is necessary, since clicking anywhere outside the dialogue also closes it.